### PR TITLE
OnError input prop function call on errors (#24)

### PR DESCRIPTION
**Pull request by andriygavriluk**
Please check out my error handling version

### CODE REVIEW

**Comment by jxom** [comment]
Looks pretty good to me. Can we perhaps update the readme to demonstrate how the error handler works?

File: src/index.js [hunk 1]
```diff
@@ -411,11 +428,16 @@ class CreditCardInput extends Component<Props, State> {
...
-  setFieldInvalid = (errorText: string) => {
+  setFieldInvalid = (errorText: string, inputName) => {
     const { invalidClassName } = this.props;
     // $FlowFixMe
     document.getElementById('field-wrapper').classList.add(invalidClassName);
     this.setState({ errorText });
+
+    if (inputName) {
+      const { onError } = this.props[`${inputName}InputProps`];
+      onError && onError(errorText);
```
Comment (jxom) [inline src/index.js]: Can we actually make the argument of `onError` an object with keys `inputName` and `error` perhaps? The consumer may want the name of the input.

**Review by jxom — APPROVED** [review]

**Comment by jxom** [comment]
Maybe we should also have an `onError` handler at the root level props as well. What do you think?

**Comment by andriygavriluk** [comment]
@jxom 

> Looks pretty good to me. Can we perhaps update the readme to demonstrate how the error handler works?

Yeah, sure. I will update it

> Maybe we should also have an `onError` handler at the root level props as well. What do you think?

I was thinking about this. It seems like I'm going to use this feature. But first, it has to be implemented :)

**Comment by andriygavriluk** [comment]
@jxom, well all changes are done - it should work well

**Comment by mmmikeal** [comment]
thank you!

**Comment by mmmikeal** [comment]
@Gorgonello hey do you mind explaining what this does?

`onError?: Function`

I see this ternary? expression quite a bit in the repo and I have no idea what it means

it is used here as well
`inputName?: string`

**Comment by andriygavriluk** [comment]
@mmmikeal, It's a flow (static type checker for js) "optional parameter". You can read more [here](https://flow.org/en/docs/types/functions/#toc-optional-parameters).

**Comment by jxom** [comment]
Thanks!

**Pull request merged**